solidity/contracts/Gravity.sol

```diff
@@ -471,13 +470,12 @@ contract Gravity is ReentrancyGuard {
 		uint256 _nonce,
 		address _newDestination
 	) public nonReentrant {
-		if(state_RevertedVouchers[_nonce].activate){
-			TransferReverted memory voucher = state_RevertedVouchers[_nonce];
-			require(voucher.destination == msg.sender, "Can only be redeemed by intended recipient");
-			state_RevertedVouchers[_nonce].activate = false;
-			state_RevertedVouchers[_nonce].amount = 0;
-			IERC20(voucher.tokenContract).safeTransfer(_newDestination, voucher.amount);
-		}
+		TransferReverted memory voucher = state_RevertedVouchers[_nonce];
+		require(voucher.destination == msg.sender, "Can only be redeemed by intended recipient");
+		require(state_RevertedVouchers[_nonce].amount > 0, "Voucher has been redeemed already");
+		// redeemVoucher
+		state_RevertedVouchers[_nonce].amount = 0;
+		IERC20(voucher.tokenContract).safeTransfer(_newDestination, voucher.amount);
```

Comment:
Looks good to me.
The activate boolean flag is redundant as the condition of executing redeemVoucher function must be true.
And the new require check of redeem amount is indeed to avoid the zero invalid value.